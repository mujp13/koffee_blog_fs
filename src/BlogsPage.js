import React, { Component } from 'react';
import Blog from './Blog';

function BlogSearch() {
  return <p>Search bar here</p>;
}

export default class BlogsPage extends Component {
  render() {
    return (
      <>
        <BlogSearch />
        <div className="blog-list">
          <Blog />
          <Blog />
          <Blog />
        </div>
      </>
    );
  }
}
