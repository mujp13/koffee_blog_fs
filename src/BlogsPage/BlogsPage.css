.blog-type {
  text-align: center;
}

.type-button {
  width: 180px;
  height: 50px;
  font-size: 150%;
}
