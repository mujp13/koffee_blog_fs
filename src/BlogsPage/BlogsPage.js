import React, { Component } from 'react';
import Blog from '../Blog/Blog';
import './BlogsPage.css';
import KoffeeBlogContext from '../KoffeeBlogContext/KoffeeBlogContext';

export default class BlogsPage extends Component {
  static contextType = KoffeeBlogContext;

  state = {
    types: ['Bean', 'Iced Coffee', 'Brewing', 'Drinks'],
    filteredType: null,
  };

  changeType = (type) => {
    this.setState({
      filteredType: type,
    });
  };

  componentDidMount() {
    this.context.loadData();
  }

  render() {
    return (
      <>
        <div className="blog-type">
          {this.state.types.map((type, index) => (
            <button key={index} className="type-button" onClick={(e) => this.changeType(type)}>
              {type}
            </button>
          ))}
          <div className="ph-clear"></div>
        </div>
        <div className="blog-list">
          {this.state.filteredType
            ? this.context.blogs
                .filter((type) => type.typeid === this.state.filteredType)
                .map((blog, index) => (
                  <Blog
                    key={index}
                    title={blog.title}
                    content={blog.content}
                    imageurl={blog.imageurl}
                    date={blog.date_published}
                  />
                ))
            : this.context.blogs.map((blog, index) => (
                <Blog
                  key={index}
                  title={blog.title}
                  content={blog.content}
                  imageurl={blog.imageurl}
                  rating={blog.rating}
                  location={blog.location}
                />
              ))}
        </div>
      </>
    );
  }
}
