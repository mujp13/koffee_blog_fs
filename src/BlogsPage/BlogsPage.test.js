import React from 'react';
import ReactDOM from 'react-dom';
import BlogsPage from './BlogsPage';
import { MemoryRouter } from 'react-router-dom';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<BlogsPage />, div);

  ReactDOM.unmountComponentAtNode(div);
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <MemoryRouter>
      <BlogsPage />
    </MemoryRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});
