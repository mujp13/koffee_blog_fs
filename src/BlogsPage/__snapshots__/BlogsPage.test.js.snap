// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="blog-type"
  >
    <button
      className="type-button"
      onClick={[Function]}
    >
      Bean
    </button>
    <button
      className="type-button"
      onClick={[Function]}
    >
      Iced Coffee
    </button>
    <button
      className="type-button"
      onClick={[Function]}
    >
      Brewing
    </button>
    <button
      className="type-button"
      onClick={[Function]}
    >
      Drinks
    </button>
    <div
      className="ph-clear"
    />
  </div>,
  <div
    className="blog-list"
  />,
]
`;
